src/utils/Rotator.ts

```diff
@@ -0,0 +1,544 @@
+   */
+  focus(el: HTMLElement) {
+    // Avoid focusing on already focused element
+    if (el && el === this.el) {
+      return;
+    }
+
+    this.el = el;
+    this.updateContainer({ forceShow: true });
+    on(this.getDocumentEl(), 'pointerdown', this.handleMouseDown);
+  }
+
+  /**
+   * Blur from element
+   */
+  blur() {
+    this.container!.style.display = 'none';
+
+    if (this.el) {
+      off(this.getDocumentEl(), 'pointerdown', this.handleMouseDown);
+      delete this.el;
+    }
+  }
+
+  /**
+   * Start rotating
+   * @param  {Event} e
+   */
+  start(ev: Event) {
+    const e = ev as PointerEvent;
+    // @ts-ignore Right or middel click
+    if (e.button !== 0) return;
+    e.preventDefault();
+    e.stopPropagation();
+    const el = this.el!;
+    const rotator = this;
+    const config = this.opts || {};
+    const attrName = 'data-' + config.prefix + 'handler';
+    const rectRotation = this.getElementRotation(el!);
+    const rect = this.getElementPos(el!, { avoidFrameZoom: true, });
+    const target = e.target as HTMLElement;
+    this.handlerAttr = target.getAttribute(attrName)!;
+
+    const mouseFetch = this.mousePosFetcher;
+    this.startPos = mouseFetch
+      ? mouseFetch(e)
+      : {
+          x: e.clientX,
+          y: e.clientY,
+        };
+
+    this.startDim = {
+      t: rect.top,
+      l: rect.left,
+      w: rect.width,
+      h: rect.height,
+      r: rectRotation,
+    };
+    this.rectDim = {
+      t: rect.top,
+      l: rect.left,
+      w: rect.width,
+      h: rect.height,
+      r: rectRotation,
+    };
+
+    const dims = this.getElementPos(el!, { avoidFrameZoom: true, avoidFrameOffset: true });
+    this.center = {
+      x: dims.left + (dims.width / 2),
+      y: dims.top + (dims.height / 2),
+    };
+
+    // Listen events
+    const docs = this.getDocumentEl();
+    this.docs = docs;
+    on(docs, 'pointermove', this.move);
+    on(docs, 'keydown', this.handleKeyDown);
+    on(docs, 'pointerup', this.stop);
+    isFunction(this.onStart) && this.onStart(e, { docs, config, el, rotator });
+    this.toggleFrames(true);
+    this.move(e);
+  }
+
+  /**
+   * While rotating
+   * @param  {Event} e
+   */
+  move(ev: PointerEvent | Event) {
+    const e = ev as PointerEvent;
+    const onMove = this.onMove;
+    const mouseFetch = this.mousePosFetcher;
+    const currentPos = mouseFetch
+      ? mouseFetch(e)
+      : {
+          x: e.clientX,
+          y: e.clientY,
+        };
+    this.currentPos = currentPos;
+
+    const dX = this.startPos!.x - this.center!.x;
+    const dY = this.startPos!.y - this.center!.y;
+    const R = Math.sqrt(dX * dX + dY * dY);
+
+    const vX = currentPos.x - this.center!.x;
+    const vY = currentPos.y - this.center!.y;
+    const magV = Math.sqrt(vX * vX + vY * vY);
+    const aX = this.center!.x + vX / magV * R;
+    const aY = this.center!.y + vY / magV * R;
+
+    this.delta = {
+      x: aX - this.center!.x,
+      y: aY - this.center!.y,
+    };
```

Comment:
```suggestion
    const dX = vX / magV * R;
    const dY = vY / magV * R;

    this.delta = {
      x: dX,
      y: dY,
    };
```

Reply:
You sure? want het zou wel dat moenten doen want met replacement is het 

`dX = (this.center!.x + vX / magV * R) - this.center!.x`

Doordat `(a + b) + c` het zelfde is als `a + (b + c)` hoef je geen haakjes te gebruiken. 

`dX = this.center!.x + vX / magV * R - this.center!.x`

Wegens dat `(a + b) - c` het zelfde is als `(a - c) + b` kan je

`dX = this.center!.x - this.center!.x + vX / magV * R`

Dus dan maakt het geen verschil?

Reply:
```suggestion

    this.delta = {
      x:  vX / magV * R,
      y:  vY / magV * R,
    };
```